python-spec/src/somacore/base.py

```diff
@@ -61,6 +68,18 @@ def metadata(self) -> MutableMapping[str, Any]:
         """
         raise NotImplementedError()
 
+    @property
+    @abc.abstractmethod
+    def mode(self) -> options.OpenMode:
+        """Returns the mode this object was opened in."""
```

Comment:
do you want to expand to match the docstring in `tiledbsoma`, which inclues "... either 'r' or 'w'."